waku/factory/internal_config.nim

```diff
@@ -32,7 +32,14 @@ proc enrConfiguration*(
   let shards: seq[uint16] =
     # no shards configured
     if conf.shards.len == 0:
-      toSeq(0 ..< conf.pubsubTopics.len).mapIt(uint16(it))
+      var shardsLocal = newSeq[uint16]()
```

Comment:
I think it can be simpler:

```
(topicsToRelayShards(conf.pubsubTopics).valueOr:
        error "failed to parse pubsubTopic", error = shardsRes.error
       @[]
).get(@[])
```
Something like that is possible. Did not tried for now.

Reply:
That looks complex to read, will try it.